File: shared/src/types/hand-history.ts
import type { CardString } from './card.js';
import type { GameType } from './game.js';
import type { Street, PlayerAction } from './hand.js';

export interface HandRecord {
  handId: string;
  handNumber: number;
  gameType: GameType;
  timestamp: number;
  blinds: { small: number; big: number };
  players: HandRecordPlayer[];
  streets: HandRecordStreet[];
  pots: HandRecordPot[];
  communityCards: CardString[];
  secondBoard?: CardString[];
  summary: HandRecordSummary;
}

export interface HandRecordPlayer {
  playerId: string;
  name: string;
  seatIndex: number;
  startingStack: number;
  holeCards?: CardString[];
  isDealer: boolean;
  isSmallBlind: boolean;
  isBigBlind: boolean;
  shownAtShowdown: boolean;
}

export interface HandRecordStreet {
  street: Street;
  boardCards: CardString[];
  actions: PlayerAction[];
}

export interface HandRecordPot {
  name: string;
  amount: number;
  winners: { playerId: string; playerName: string; amount: number }[];
  winningHand?: string;
}

export interface HandRecordSummary {
  results: { playerId: string; playerName: string; netChips: number }[];
}

File: ios/PokerSofta/Sources/Models/HandHistory.swift
import Foundation

struct HandRecord: Codable, Identifiable {
    let handId: String
    let handNumber: Int
    let gameType: GameType
    let timestamp: Double
    let blinds: Blinds
    let players: [HandRecordPlayer]
    let streets: [HandRecordStreet]
    let pots: [HandRecordPot]
    let communityCards: [CardString]
    let secondBoard: [CardString]?
    let summary: HandRecordSummary

    var id: String { handId }

    struct Blinds: Codable {
        let small: Double
        let big: Double
    }
}

struct HandRecordPlayer: Codable {
    let playerId: String
    let name: String
    let seatIndex: Int
    let startingStack: Double
    let holeCards: [CardString]?
    let isDealer: Bool
    let isSmallBlind: Bool
    let isBigBlind: Bool
    let shownAtShowdown: Bool
}

struct HandRecordStreet: Codable {
    let street: Street
    let boardCards: [CardString]
    let actions: [PlayerAction]
}

struct HandRecordPot: Codable {
    let name: String
    let amount: Double
    let winners: [PotWinner]
    let winningHand: String?

    struct PotWinner: Codable {
        let playerId: String
        let playerName: String
        let amount: Double
    }
}

struct HandRecordSummary: Codable {
    let results: [PlayerResult]

    struct PlayerResult: Codable {
        let playerId: String
        let playerName: String
        let netChips: Double
    }
}
